packages/core/test/http/apiLogger.test.ts

```diff
@@ -0,0 +1,201 @@
+import {
+  HttpRequest,
+  HttpResponse,
+  Level,
+  LoggingOptions,
+} from '../../src/coreInterfaces';
+import { ConsoleLogger } from '../../src/logger/defaultLogger';
+import { ApiLogger } from '../../src/logger/apiLogger';
+import { callHttpInterceptors } from '../../src/http/httpInterceptor';
+import { NullLogger } from '../../src/logger/nullLogger';
+
+let loggerSpy;
+beforeEach(() => {
+  // Reset the spy on console.log() before each test
+  loggerSpy = jest.spyOn(console, 'log').mockImplementation();
+});
+
+afterEach(() => {
+  // Restore the original implementation of console.log() after each test
+  loggerSpy.mockRestore();
+});
+
+describe('APILogger with ConsoleLogging', () => {
+  it('should log req and response body, headers with include header filters', async () => {
```

Comment:
Each unit test must test one small behavior or requirement at a time. When the unit test statements are read together, they must read like a requirements list for the class or feature that you are testing out. In this case, you are testing whether the logger can log at all, the logging format by default, the ability to log requests, the ability to log responses, the ability to log body, the ability to log headers, and the ability to log with filters applied. When this unit test fails, you won't be told which of these abilities/behaviors failed to work just by the status of the unit test which is a sign that your unit test is not focused on testing one thing.

Reply:
added unit test to check one behavior at a time